File: hospital-appointment-backend/src/main/java/com/hospital/appointment/entity/Appointment.java
package com.hospital.appointment.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 挂号记录实体类
 * 改造自预约记录(reservationrecord)
 */
@Data
@TableName("appointment")
public class Appointment implements Serializable {
    
    @TableId(type = IdType.INPUT)
    private String appointmentID;
    
    /** 患者ID */
    private Long patientID;
    
    /** 排班ID */
    private Long scheduleID;
    
    /** 医生ID */
    private Long doctorID;
    
    /** 科室ID */
    private Long departmentID;
    
    /** 预约就诊日期 */
    private LocalDate appointmentDate;
    
    /** 时段：1-上午 2-下午 3-夜诊 */
    private Integer timeSlot;
    
    /** 就诊序号 */
    private Integer appointmentNumber;
    
    /** 主诉/病情描述 */
    private String chiefComplaint;
    
    /** 挂号状态：0-待支付 1-已预约 2-已签到 3-就诊中 4-已完成 5-已取消 6-已退号 7-已爽约 */
    private Integer appointmentStatus;
    
    /** 支付状态：0-待支付 1-已支付 2-已退款 */
    private Integer paymentStatus;
    
    /** 支付金额 */
    private BigDecimal paymentAmount;
    
    /** 支付方式 */
    private String paymentMethod;
    
    /** 支付时间 */
    private LocalDateTime paymentTime;
    
    /** 取消原因 */
    private String cancelReason;
    
    /** 取消时间 */
    private LocalDateTime cancelTime;
    
    /** 是否已评价：0-否 1-是 */
    private Integer isReviewed;
    
    /** 创建时间 */
    private LocalDateTime createTime;
    
    /** 非数据库字段：医生姓名 */
    @TableField(exist = false)
    private String doctorName;
    
    /** 非数据库字段：科室名称 */
    @TableField(exist = false)
    private String departmentName;
    
    /** 非数据库字段：患者姓名 */
    @TableField(exist = false)
    private String patientName;
    
    /** 非数据库字段：出诊日期-时段格式 */
    @TableField(exist = false)
    private String scheduleTimeStr;
}

File: hospital-appointment-frontend/src/api/hospital/appointment.ts
/**
 * 挂号预约 API
 */

import request from '../request'

export interface Appointment {
  appointmentID: string
  patientID: number
  patientName?: string
  scheduleID: number
  doctorID: number
  doctorName?: string
  departmentID: number
  departmentName?: string
  appointmentDate: string
  timeSlot: number
  appointmentNumber: number
  chiefComplaint?: string
  appointmentStatus: number
  paymentStatus: number
  paymentAmount: number
  paymentMethod?: string
  paymentTime?: string
  cancelReason?: string
  cancelTime?: string
  isReviewed: number
  createTime: string
  scheduleStartTime?: string
  scheduleEndTime?: string
}

export interface CreateAppointmentData {
  patientID: number
  scheduleID: number
  appointmentDate: string
  timeSlot: number
  chiefComplaint?: string
}

export const appointmentAPI = {
  /**
   * 获取所有挂号记录（管理员）
   */
  getAll: () => {
    return request.get<Appointment[]>('/hospital/appointment/admin/all')
  },

  /**
   * 创建挂号
   */
  create: (data: CreateAppointmentData) => {
    return request.post<string>('/hospital/appointment', data)
  },

  /**
   * 获取患者的所有挂号记录
   */
  getByPatient: (patientId: number, status?: number) => {
    return request.get<Appointment[]>(`/hospital/appointment/patient/${patientId}`, {
      params: status !== undefined ? { status } : {}
    })
  },

  /**
   * 获取医生的所有挂号记录
   */
  getByDoctor: (doctorId: number, status?: number) => {
    return request.get<Appointment[]>(`/hospital/appointment/doctor/${doctorId}`, {
      params: status !== undefined ? { status } : {}
    })
  },

  /**
   * 获取挂号详情
   */
  getById: (id: string) => {
    return request.get<Appointment>(`/hospital/appointment/${id}`)
  },

  /**
   * 取消挂号
   */
  cancel: (id: string, reason?: string) => {
    return request.put<string>(`/hospital/appointment/${id}/cancel`, null, {
      params: { reason }
    })
  },

  /**
   * 签到
   */
  checkIn: (id: string) => {
    return request.put<string>(`/hospital/appointment/${id}/checkin`)
  },

  /**
   * 支付成功回调
   */
  paySuccess: (id: string, paymentMethod: string) => {
    return request.put<string>(`/hospital/appointment/${id}/pay`, null, {
      params: { paymentMethod }
    })
  },

  /**
   * 确认就诊
   */
  confirm: (id: string) => {
    return request.put<string>(`/hospital/appointment/${id}/confirm`)
  },

  /**
   * 完成就诊
   */
  finish: (id: string) => {
    return request.put<string>(`/hospital/appointment/${id}/finish`)
  },

  /**
   * 获取当日挂号列表
   */
  getToday: (doctorId: number) => {
    return request.get<Appointment[]>('/hospital/appointment/today', { params: { doctorId } })
  },

  /**
   * 获取明日挂号数
   */
  getTomorrowCount: (doctorId: number) => {
    return request.get<number>('/hospital/appointment/tomorrow/count', { params: { doctorId } })
  }
}
